### PR TITLE
block_index --> base_element in assembly.cc

### PR DESCRIPTION
**Pull request by rrgrove6**
Attempting to fix a bug

@tjhei should I rename block_idx (see below)? Or was the code correct as is here?

Wait for test results before merge.

### CODE REVIEW

File: source/simulator/assembly.cc [hunk 1]
```diff
@@ -1367,7 +1367,7 @@ namespace aspect
           Assert (false, ExcNotImplemented());
       }
 
-    const unsigned int block_idx = advection_field.block_index(introspection);
+    const unsigned int block_idx = advection_field.base_element(introspection);
```
Comment (tjhei) [inline source/simulator/assembly.cc]: Did you run your PR? This needs to be ```block_index```! See where it is used? It accesses a matrix block using it.

File: source/simulator/assembly.cc [hunk 2]
```diff
@@ -1660,7 +1660,7 @@ namespace aspect
       computing_timer.enter_section ("   Assemble temperature system");
     else
       computing_timer.enter_section ("   Assemble composition system");
-    const unsigned int block_idx = advection_field.block_index(introspection);
+    const unsigned int block_idx = advection_field.base_element(introspection);
```
Comment (tjhei) [inline source/simulator/assembly.cc]: nope, see above

**Comment by rrgrove6** [comment]
@jdannberg merge this without question please (;

**Comment by tjhei** [comment]
:+1:

**Comment by tjhei** [comment]
somebody else please take a look and merge (I am biased)

**Comment by jdannberg** [comment]
Can you add a test case that didn't work before, but does now?
(like graphical output of the entropy viscosity with a direct solver?)

**Comment by rrgrove6** [comment]
I'm not sure what did or didn't work before.. but maybe?

**Comment by tjhei** [comment]
visualize the entropy viscosity of a compositional field using get_artificial_viscosity() should make it fail. Do you remember what test crashed with the "index 4" error? You can use that one.

**Comment by rrgrove6** [comment]
So, the test works before and after the change, so I suppose we need to be more clever to break it. I also tried turning the direct solver off and it only fails with  different values, but not the [0, 4[ bug we saw.

**Comment by bangerth** [comment]
What happened to the actual code changes in this PR? There are no changes to assembly.cc any more...

**Comment by tjhei** [comment]
Here is a test that crashes on mainline (the message also contains the fix): https://github.com/tjhei/aspect/commit/4beec3c323f53e05e457728e416ebe2ccf4d65d9

Ryan, please update.

**Comment by tjhei** [comment]
Can you please fix the whitespace in the test output? You can access the correctly formatted files if you click on "raw", for example [1]. Instead of copying the files, a better way would have been to cherry-pick my commit onto your branch. I will show you the next time I see you in person. :-)

Oh, also squash those four commits. The old ones are no longer useful now.

[1]: https://raw.githubusercontent.com/tjhei/aspect/4beec3c323f53e05e457728e416ebe2ccf4d65d9/tests/artificial_postprocess/screen-output

**Comment by bangerth** [comment]
@rrgrove6 ?

**Comment by rrgrove6** [comment]
Good to go, I think.

**Pull request merged**